Basic/ClientDriven/Assets/Scripts/ClientDrivenNetworkTransform.cs

```diff
@@ -0,0 +1,44 @@
+using Unity.Multiplayer.Samples.Utilities.ClientAuthority;
+using UnityEngine;
+
+/// <summary>
+/// Component inheriting from <see cref="ClientNetworkTransform"/>, where server-driven player position changes are
+/// applied to the owning client.
+/// </summary>
+/// <remarks>
+/// Handling movement inside this component's OnNetworkSpawn method only ensures the mitigation of race condition issues
+/// arising due to the execution order of other NetworkBehaviours' OnNetworkSpawn methods.
+/// </remarks>
+[RequireComponent(typeof(ServerPlayerMove))]
+[DisallowMultipleComponent]
+public class ClientDrivenNetworkTransform : ClientNetworkTransform
+{
+    [SerializeField]
+    ServerPlayerMove m_ServerPlayerMove;
+
+    public override void OnNetworkSpawn()
+    {
+        base.OnNetworkSpawn();
+
+        if (IsClient && IsOwner)
+        {
+            OnValueChanged(Vector3.zero, m_ServerPlayerMove.spawnPosition.Value);
+            m_ServerPlayerMove.spawnPosition.OnValueChanged += OnValueChanged;
```

Comment:
ServerPlayerMove is a server-side script. Why are we initializing it on every local player? It should be initialized only on the server. Its https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/56154f106d15ec0b096f7879d2d28db471dffd5e/Basic/ClientDriven/Assets/Scripts/ServerPlayerMove.cs#L21 will also automatically disable this line on clients, making it useless.

Am I missing something?

Reply:
Right, so we did that previously, but due to the regressions documented inside ServerPlayerMove.OnServerSpawnPlayer(), the recommended pattern for initializing a player's position should be client-side, through a NetworkVariable modified by the server.

Per @NoelStephensUnity :

> I typically recommend the client obtains the teleport position via NetworkVariable set by the server or by the client grabbing it from a set of spawn points (i.e. like the server is doing in ClientDriven) and then teleporting the player to that position when the player spawns on the owner side. While the NetworkObject does synchronize its position there are a number of things that could cause this to fail with an owner authoritative NT.

And so that's what this PR is doing. ServerPlayerMove.spawnPosition is set on OnNetworkSpawn server-side, and the client consumes and applies that position on OnNetworkSpawn.